### PR TITLE
test: Added tests for /etxns/:id to platform migration in view expense page

### PR DESCRIPTION
**Pull request by OmkarJ13**
### <samp>🤖 Generated by Copilot at ce13767</samp>

The pull request updates the mock data, the view-expense component tests, and the expense model to support a new feature for matching corporate card transactions. The feature requires displaying the name of the corporate card user on the expense details page.

### <samp>🤖 Generated by Copilot at ce13767</samp>

> _To show who the card owner is_
> _We added a field with their name_
> _The mock data changed_
> _The tests rearranged_
> _And `receiptDetails` methods became lame_

### <samp>🤖 Generated by Copilot at ce13767</samp>

*  Add corporate_card_user_full_name field to MatchedCorporateCardTransaction interface to store the name of the user who owns the corporate card ([link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-57a2d67f855d51e1a242e5077eea76af56c28517c98ae770f47d197b126946b8L137-R138))
*  Change the expenseData object and its fields in mock-data/platform/v1/expense.data.ts to different values for testing purposes ([link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bL12-R21), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bL27-R54), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bL58-R86), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bL276-R96), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bL286-R133), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bL305-R146), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bL320-R209), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bL348-R223))
*  Replace fileService.findByTransactionId method by fileService.getReceiptsDetails method in view-expense.page.spec.ts and view-expense.page.ts to get the receipt details of an expense ([link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feR45-R50), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL83-R86), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feR121-R122), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL433-R471), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL446-R479), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL459-R492), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL474-R510), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL490-R524), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL499-R537), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL515-R553), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL531-R577), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL547-R609))
*  Import and declare ApproverExpensesService and SpenderExpensesService as jasmine spy objects in view-expense.page.spec.ts to mock the services that fetch the expense details based on the user role ([link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL26-R39), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feR73-R74), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feR201-R208), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feR247-R248), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL595-R634))
*  Change the setPaymentModeandIcon method in view-expense.page.spec.ts to use the new expenseData object and its fields to test the payment mode and icon of an expense ([link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL354-R380), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL368-R393), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL377-R399), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL385-R419), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL399-R438))
*  Change the getFlagStatus method in view-expense.page.spec.ts to use the new expenseData object and its fields to test the flag status of an expense ([link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL613-R647), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL635-R680), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL653-R695), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL666-R701), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL672-R714))
*  Change the isAmountCapped method in view-expense.page.spec.ts to use the new expenseData object and its fields to test the amount capped condition of an expense ([link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL684-R720), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL691-R733), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL708-R755), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL726-R773))
*  Change the isCriticalPolicyViolated method in view-expense.page.spec.ts to use the new expenseData object and its fields to test the critical policy violation condition of an expense ([link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL795-R860))
*  Remove the getReceiptExtension method and the getReceiptDetails method from view-expense.page.spec.ts as they are no longer used in the component ([link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL585-R620), [link](https://github.com/fylein/fyle-mobile-app/pull/2548/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL903-L971))

## Clickup
https://app.clickup.com/t/8678z1xhp

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/models/platform/v1/expense.model.ts [hunk 1]
```diff
@@ -134,8 +134,8 @@ export interface MatchedCorporateCardTransaction {
   corporate_card_id: string;
   corporate_card_number: string;
   masked_corporate_card_number: string;
+  corporate_card_user_full_name: string;
```
Comment (OmkarJ13) [inline src/app/core/models/platform/v1/expense.model.ts:137]: Ref: https://fylein.slack.com/archives/C05GR21GAER/p1699436281560379

File: src/app/fyle/view-expense/view-expense.page.spec.ts [hunk 2]
```diff
@@ -900,75 +928,6 @@ describe('ViewExpensePage', () => {
     });
   });
 
-  describe('getReceiptExtension():', () => {
```
Comment (OmkarJ13) [inline src/app/fyle/view-expense/view-expense.page.spec.ts:903]: Not required anymore

File: src/app/core/models/platform/v1/expense.model.ts (hunk 1 above)
Comment (SowmyaFyle) [reply src/app/core/models/platform/v1/expense.model.ts:137]: have we removed `cardholder_name` key as well?

**Review by SowmyaFyle — APPROVED** [review]

File: src/app/fyle/view-expense/view-expense.page.spec.ts [hunk 3]
```diff
@@ -351,60 +366,75 @@ describe('ViewExpensePage', () => {
...
+        },
       };
-      component.expense$ = of(mockExchangeRateExpData);
-      component.setPaymentModeandIcon(mockExchangeRateExpData);
+
+      component.expense$ = of(mockExpense);
+      component.setPaymentModeandIcon(mockExpense);
       component.expense$.subscribe((res) => {
-        expect(res.tx_skip_reimbursement).toBeTrue();
+        expect(res.is_reimbursable).toBeFalse();
```
Comment (hlkavya0213) [inline src/app/fyle/view-expense/view-expense.page.spec.ts:418]: better name for res

File: src/app/fyle/view-expense/view-expense.page.spec.ts [hunk 4]
```diff
@@ -351,60 +366,75 @@ describe('ViewExpensePage', () => {
...
+        is_reimbursable: true,
       };
-      component.expense$ = of(mockExchangeRateExpData);
-      component.setPaymentModeandIcon(mockExchangeRateExpData);
+
+      component.expense$ = of(mockExpense);
+      component.setPaymentModeandIcon(mockExpense);
       component.expense$.subscribe((res) => {
-        expect(res.source_account_type).toEqual('PERSONAL_ACCOUNT');
+        expect(res.source_account.type).toEqual(AccountType.PERSONAL_CASH_ACCOUNT);
```
Comment (hlkavya0213) [inline src/app/fyle/view-expense/view-expense.page.spec.ts:437]: same as above fix it everywhere

**Review by hlkavya0213 — APPROVED** [review]
minor comments

File: src/app/core/models/platform/v1/expense.model.ts (hunk 1 above)
Comment (OmkarJ13) [reply src/app/core/models/platform/v1/expense.model.ts:137]: yes

**Review by abhishek1234321 — APPROVED** [review]

**Pull request merged**